<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tumour microenvironment cell estimation (ConsensusTME)</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Aneuploidy inference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.research.ieo.it/research-and-technology/technological-units/data-science-unit/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tumour microenvironment cell estimation
(ConsensusTME)</h1>
<h4 class="date">05 June, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>casper_ss_ma/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/samples_metadata.csv
</td>
<td style="text-align:left;">
../data/samples_metadata.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/data.Rdata
</td>
<td style="text-align:left;">
../data/data.Rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/consensusTME.csv
</td>
<td style="text-align:left;">
../data/consensusTME.csv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatree2840d6ddedbb1a4028f6bd38cea98c903072c51ftargetblank2840d6da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/2840d6ddedbb1a4028f6bd38cea98c903072c51f" target="_blank">2840d6d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatree2840d6ddedbb1a4028f6bd38cea98c903072c51ftargetblank2840d6da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/2840d6ddedbb1a4028f6bd38cea98c903072c51f" target="_blank">2840d6d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/01_degs_go.Rmd
    Untracked:  data/
    Untracked:  support_code.R

Unstaged changes:
    Modified:   analysis/00_casper_analysis.Rmd
    Modified:   casper_ss_ma.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/02_deconvolution.Rmd</code>) and
HTML (<code>docs/02_deconvolution.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/2840d6ddedbb1a4028f6bd38cea98c903072c51f/analysis/02_deconvolution.Rmd" target="_blank">2840d6d</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
<td>
workflowr::wflow_publish("analysis/02_deconvolution.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/dimadatascience/casper_ss_ma/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/02_deconvolution.html" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/826ff33e06d7c3af331493777e7f6f1f3840879d/analysis/02_deconvolution.Rmd" target="_blank">826ff33</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
<td>
workflowr::wflow_publish("analysis/02_deconvolution.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)</code></pre>
<div id="consensustme" class="section level2">
<h2>ConsensusTME</h2>
<pre class="r"><code>library(RSQLite)
library(ConsensusTME)

# ConsensusTME::cancerAll # BRCA</code></pre>
<p>The ConsensusTME package provides a integrative tool for R that uses
a consensus approach to generating cancer specific signatures for
multiple cell types found within the tumour microenvironment.</p>
<p>These consensus gene sets are then used within a ssGSEA framework to
provide normalised enrichment scores for each of the cell types
representing the relative abundance of cell types across multiple
samples.</p>
<p>It is important to note that results generated in this way will
provide quantification of cell types that are relative across samples
rather than across cell types.</p>
<p>returns estimation of cell type abundance for each sample in the bulk
tumour gene expression matrix. Immune Score represents the overall level
of immune infiltration for each sample.</p>
<pre class="r"><code># sample_info table
sample_info = read.table(&quot;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/samples_metadata.csv&quot;, header=T, sep = &#39;,&#39;, row.names = 1)
rownames(sample_info) &lt;- sample_info$samplename
sample_info$condition &lt;- as.factor(sample_info$condition)
sample_info$batch &lt;- as.factor(sample_info$batches)


# annotation
load(&#39;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/data.Rdata&#39;)
#head(data) # TPM counts

# convert ENS gene id to Gene symbol
# gene symbol x patients # Matrix is a numerical matrix with gene symbol as rownames
matrix=data
genes=sub(&quot;\\..*&quot;, &quot;&quot;, rownames(matrix))
length(genes)</code></pre>
<pre><code>[1] 48643</code></pre>
<pre class="r"><code>length(unique(genes))</code></pre>
<pre><code>[1] 48643</code></pre>
<pre class="r"><code>gene_map &lt;- setNames(annotation$GeneSymbol, annotation$Gene)
symbols &lt;- gene_map[genes]
sum(is.na(symbols))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sum(duplicated(symbols[!is.na(symbols)]))</code></pre>
<pre><code>[1] 12070</code></pre>
<pre class="r"><code>summary_df &lt;- data.frame(
  Total = length(symbols),
  Unique = length(unique(symbols)),
  NAs = sum(is.na(symbols)),
  Duplicated = sum(duplicated(symbols[!is.na(symbols)])),
  Empty = sum(symbols == &quot;&quot; | symbols == &quot; &quot;, na.rm = TRUE)
)
print(summary_df)</code></pre>
<pre><code>  Total Unique NAs Duplicated Empty
1 48643  36573   0      12070 12064</code></pre>
<pre class="r"><code>symbols &lt;- make.unique(as.character(symbols))
matrix=apply(matrix, 2, as.numeric)
rownames(matrix)=symbols

matrix=matrix[!is.na(rownames(matrix)), ]
matrix=matrix[rownames(matrix)!=&quot; &quot;, ]
matrix=matrix[rownames(matrix)!=&quot;&quot;, ]

matrix[1:10,1:10]</code></pre>
<pre><code>       S58505_HR_plus_7 S58506_HR_plus_15 S58507_HR_plus_12 S58508_HR_plus_5
DPM1             15.916            13.281            15.728           15.069
SCYL3            14.932            18.384            19.453           17.914
FIRRM            11.570            11.798             9.876           10.387
FGR               3.312             3.145             4.254            3.592
CFH              27.985            43.312            43.208           35.365
FUCA2            15.597            14.116            15.654           16.926
GCLC             27.593            18.909            22.520           45.403
NFYA             29.826            33.545            29.806           28.779
STPG1             1.920             3.094             2.464            2.959
NIPAL3           23.345            17.738            17.147           14.233
       S58509_TNBC_14 S58510_HR_plus_11 S58511_HR_plus_16 S58512_HR_plus_6
DPM1           33.159            31.519            14.970           15.930
SCYL3           9.918            12.143             9.062           17.188
FIRRM          10.864             9.063             7.962           10.645
FGR             6.579             1.621             8.091            2.126
CFH            32.746            14.032            27.941           16.921
FUCA2          34.472            13.416            10.453           10.133
GCLC           27.022             9.399            17.179           29.156
NFYA           30.165            29.682            19.129           25.339
STPG1           2.652             1.644             2.037            2.873
NIPAL3         15.359             8.391             9.671           15.605
       S58513_HR_plus_14 S58514_HR_plus_8
DPM1              21.426           11.599
SCYL3             14.663            7.321
FIRRM              9.546            4.911
FGR                2.583            1.736
CFH               23.358           37.848
FUCA2             17.493            9.405
GCLC              26.855           20.424
NFYA              18.850           15.604
STPG1              2.724            0.748
NIPAL3            21.230            9.370</code></pre>
<pre class="r"><code>consensus=ConsensusTME::consensusTMEAnalysis(matrix, cancerType = &#39;BRCA&#39;, )</code></pre>
<pre><code>Producing ConsensusTME Estimates Using The Following Parameters:
 Statistical Framework: &quot;ssgsea&quot;
 Gene Sets For Cancer Type: &quot;BRCA&quot;
 Sample Size: 149</code></pre>
<pre><code>Estimating ssGSEA scores for 19 gene sets.
[1] &quot;Calculating ranks...&quot;
[1] &quot;Calculating absolute values from ranks...&quot;
  |                                                                              |                                                                      |   0%  |                                                                              |                                                                      |   1%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   4%  |                                                                              |===                                                                   |   5%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  17%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |===============                                                       |  21%  |                                                                              |================                                                      |  22%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |=================                                                     |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |===================                                                   |  28%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |=======================                                               |  34%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |==========================                                            |  38%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  40%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |===============================                                       |  44%  |                                                                              |===============================                                       |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |=================================                                     |  48%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  54%  |                                                                              |=======================================                               |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  57%  |                                                                              |========================================                              |  58%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  62%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  64%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |===============================================                       |  68%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |=====================================================                 |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |======================================================                |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  80%  |                                                                              |========================================================              |  81%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |===========================================================           |  84%  |                                                                              |===========================================================           |  85%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |===============================================================       |  91%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================|  99%  |                                                                              |======================================================================| 100%

[1] &quot;Normalizing...&quot;</code></pre>
<pre class="r"><code>consensus=t(consensus)
consensus=merge(consensus, sample_info, by = &quot;row.names&quot;, all = TRUE )</code></pre>
</div>
<div
id="correlation-between-tme-enrichment-and-aneuploidy-score-from-casper"
class="section level1">
<h1>Correlation between TME enrichment and aneuploidy score (from
CaSPeR)</h1>
<pre class="r"><code>library(ggplot2)
library(ggpubr)    # For stat_cor()
library(patchwork) # For layout

target_var &lt;- &quot;aneuploidy_score&quot;
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots &lt;- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = &quot;condition&quot;)) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;blue&quot;) +
    stat_cor(method = &quot;pearson&quot;, label.x.npc = &quot;left&quot;, label.y.npc = &quot;top&quot;, size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10)
})

# Combine into a grid
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-4-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-4-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-between-tme-enrichment-and-cd82-gene-expression"
class="section level1">
<h1>Correlation between TME enrichment and CD82 gene expression</h1>
<pre class="r"><code>target_var &lt;- &quot;ENSG00000085117&quot;
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots &lt;- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = &quot;condition&quot;)) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;blue&quot;) +
    stat_cor(method = &quot;pearson&quot;, label.x.npc = &quot;left&quot;, label.y.npc = &quot;top&quot;, size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10)
})

# Combine into a grid
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-5-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-5-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="boxplot-of-tme-enrichment-across-conditions"
class="section level1">
<h1>Boxplot of TME enrichment across conditions</h1>
<pre class="r"><code>df=consensus
df=df[!is.na(df$ENSG00000085117), ]
df$CD82_levels=ifelse(df$ENSG00000085117&gt;median(df$ENSG00000085117), &#39;high-CD82&#39;, &#39;low-CD82&#39;)
df$aneuploidy_score_levels=ifelse(df$aneuploidy_score&gt;median(df$aneuploidy_score), &#39;high-aneuploidy_score&#39;, &#39;low-aneuploidy_score&#39;)


condition_col &lt;- &quot;condition&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-6-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-6-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="boxplot-of-tme-enrichment-across-cd82-gene-levels-stratified-by-median"
class="section level1">
<h1>Boxplot of TME enrichment across CD82 gene levels (stratified by
median)</h1>
<pre class="r"><code>condition_col &lt;- &quot;CD82_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-7-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-7-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="boxplot-of-tme-enrichment-across-aneuploidy-score-levels-stratified-by-median"
class="section level1">
<h1>Boxplot of TME enrichment across aneuploidy score levels (stratified
by median)</h1>
<pre class="r"><code>condition_col &lt;- &quot;aneuploidy_score_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-8-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-8-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="boxplot-of-tme-enrichment-across-cd82-expression-categories-by-ihc-in-tnbc"
class="section level1">
<h1>Boxplot of TME enrichment across CD82 expression categories (by IHC)
in TNBC</h1>
<pre class="r"><code>condition_col &lt;-&quot;CATEGORY.1.&quot; # &quot;CATEGORY.1.&quot; #&quot;CD82_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df[df$condition==&quot;TNBC&quot;,], aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-9-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-9-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="boxplot-of-tme-enrichment-across-aneuploidy-score-levels-in-tnbc"
class="section level1">
<h1>Boxplot of TME enrichment across aneuploidy score levels in
TNBC</h1>
<pre class="r"><code>condition_col &lt;- &quot;aneuploidy_score_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df[df$condition==&quot;TNBC&quot;,], aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-10-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-10-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>condition_col &lt;- &quot;aneuploidy_score_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = &#39;condition&#39;)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-11-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/150ebe1adaf2ac5daaafe27b6696a3d8b02f9dd1/docs/figure/02_deconvolution.Rmd/unnamed-chunk-11-1.png" target="_blank">150ebe1</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#write.csv(consensus, &#39;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/consensusTME.csv&#39;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.3.0         ggpubr_0.6.0            ggplot2_3.5.2          
[4] ConsensusTME_0.0.1.9000 RSQLite_2.3.4          

loaded via a namespace (and not attached):
  [1] DBI_1.2.3                   bitops_1.0-9               
  [3] GSEABase_1.62.0             rlang_1.1.2                
  [5] magrittr_2.0.3              git2r_0.32.0               
  [7] matrixStats_1.1.0           compiler_4.3.1             
  [9] mgcv_1.9-0                  DelayedMatrixStats_1.22.6  
 [11] png_0.1-8                   vctrs_0.6.3                
 [13] stringr_1.5.1               pkgconfig_2.0.3            
 [15] crayon_1.5.3                fastmap_1.2.0              
 [17] backports_1.5.0             XVector_0.40.0             
 [19] labeling_0.4.3              promises_1.3.2             
 [21] rmarkdown_2.29              graph_1.78.0               
 [23] purrr_1.0.4                 bit_4.0.5                  
 [25] xfun_0.52                   zlibbioc_1.46.0            
 [27] cachem_1.1.0                beachmat_2.16.0            
 [29] GenomeInfoDb_1.36.4         jsonlite_2.0.0             
 [31] blob_1.2.4                  later_1.4.2                
 [33] rhdf5filters_1.12.1         DelayedArray_0.26.7        
 [35] Rhdf5lib_1.22.1             BiocParallel_1.34.2        
 [37] broom_1.0.8                 irlba_2.3.5.1              
 [39] parallel_4.3.1              R6_2.6.1                   
 [41] bslib_0.9.0                 stringi_1.8.7              
 [43] RColorBrewer_1.1-3          car_3.1-3                  
 [45] GenomicRanges_1.52.1        jquerylib_0.1.4            
 [47] Rcpp_1.0.12                 SummarizedExperiment_1.30.2
 [49] knitr_1.50                  GSVA_1.48.3                
 [51] IRanges_2.34.1              splines_4.3.1              
 [53] httpuv_1.6.16               Matrix_1.6-1               
 [55] tidyselect_1.2.1            rstudioapi_0.17.1          
 [57] dichromat_2.0-0.1           abind_1.4-8                
 [59] yaml_2.3.10                 codetools_0.2-19           
 [61] lattice_0.21-8              tibble_3.2.1               
 [63] Biobase_2.60.0              withr_3.0.2                
 [65] KEGGREST_1.40.1             evaluate_1.0.3             
 [67] Biostrings_2.68.1           pillar_1.10.2              
 [69] carData_3.0-5               MatrixGenerics_1.12.3      
 [71] whisker_0.4.1               stats4_4.3.1               
 [73] generics_0.1.4              rprojroot_2.0.4            
 [75] RCurl_1.98-1.14             S4Vectors_0.38.2           
 [77] sparseMatrixStats_1.12.2    scales_1.4.0               
 [79] xtable_1.8-4                glue_1.8.0                 
 [81] tools_4.3.1                 ScaledMatrix_1.8.1         
 [83] ggsignif_0.6.4              annotate_1.78.0            
 [85] fs_1.6.6                    XML_3.99-0.16              
 [87] rhdf5_2.44.0                grid_4.3.1                 
 [89] tidyr_1.3.1                 AnnotationDbi_1.62.2       
 [91] SingleCellExperiment_1.22.0 nlme_3.1-162               
 [93] GenomeInfoDbData_1.2.10     BiocSingular_1.16.0        
 [95] HDF5Array_1.28.1            Formula_1.2-5              
 [97] cli_3.6.2                   rsvd_1.0.5                 
 [99] workflowr_1.7.1             S4Arrays_1.0.6             
[101] dplyr_1.1.3                 gtable_0.3.6               
[103] rstatix_0.7.2               sass_0.4.10                
[105] digest_0.6.37               BiocGenerics_0.46.0        
[107] farver_2.1.2                memoise_2.0.1              
[109] htmltools_0.5.8.1           lifecycle_1.0.4            
[111] httr_1.4.7                  bit64_4.0.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
