<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tumour microenvironment cell estimation (ConsensusTME)</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Aneuploidy inference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.research.ieo.it/research-and-technology/technological-units/data-science-unit/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tumour microenvironment cell estimation
(ConsensusTME)</h1>
<h4 class="date">05 June, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>casper_ss_ma/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/samples_metadata.csv
</td>
<td style="text-align:left;">
../data/samples_metadata.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/data.Rdata
</td>
<td style="text-align:left;">
../data/data.Rdata
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/consensusTME.csv
</td>
<td style="text-align:left;">
../data/consensusTME.csv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatree826ff33e06d7c3af331493777e7f6f1f3840879dtargetblank826ff33a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/826ff33e06d7c3af331493777e7f6f1f3840879d" target="_blank">826ff33</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatree826ff33e06d7c3af331493777e7f6f1f3840879dtargetblank826ff33a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/826ff33e06d7c3af331493777e7f6f1f3840879d" target="_blank">826ff33</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/01_degs_go.Rmd
    Untracked:  data/
    Untracked:  support_code.R

Unstaged changes:
    Modified:   analysis/00_casper_analysis.Rmd
    Modified:   casper_ss_ma.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/02_deconvolution.Rmd</code>) and
HTML (<code>docs/02_deconvolution.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/826ff33e06d7c3af331493777e7f6f1f3840879d/analysis/02_deconvolution.Rmd" target="_blank">826ff33</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-05
</td>
<td>
workflowr::wflow_publish("analysis/02_deconvolution.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)</code></pre>
<div id="consensustme" class="section level2">
<h2>ConsensusTME</h2>
<pre class="r"><code>library(RSQLite)
library(ConsensusTME)

# ConsensusTME::cancerAll # BRCA</code></pre>
<p>The ConsensusTME package provides a integrative tool for R that uses
a consensus approach to generating cancer specific signatures for
multiple cell types found within the tumour microenvironment.</p>
<p>These consensus gene sets are then used within a ssGSEA framework to
provide normalised enrichment scores for each of the cell types
representing the relative abundance of cell types across multiple
samples.</p>
<p>It is important to note that results generated in this way will
provide quantification of cell types that are relative across samples
rather than across cell types.</p>
<pre class="r"><code># sample_info table
sample_info = read.table(&quot;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/samples_metadata.csv&quot;, header=T, sep = &#39;,&#39;, row.names = 1)
rownames(sample_info) &lt;- sample_info$samplename
sample_info$condition &lt;- as.factor(sample_info$condition)
sample_info$batch &lt;- as.factor(sample_info$batches)


# annotation
load(&#39;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/data.Rdata&#39;)
head(data) # TPM counts</code></pre>
<pre><code>                S58505_HR_plus_7 S58506_HR_plus_15 S58507_HR_plus_12
ENSG00000000419           15.916            13.281            15.728
ENSG00000000457           14.932            18.384            19.453
ENSG00000000460           11.570            11.798             9.876
ENSG00000000938            3.312             3.145             4.254
ENSG00000000971           27.985            43.312            43.208
ENSG00000001036           15.597            14.116            15.654
                S58508_HR_plus_5 S58509_TNBC_14 S58510_HR_plus_11
ENSG00000000419           15.069         33.159            31.519
ENSG00000000457           17.914          9.918            12.143
ENSG00000000460           10.387         10.864             9.063
ENSG00000000938            3.592          6.579             1.621
ENSG00000000971           35.365         32.746            14.032
ENSG00000001036           16.926         34.472            13.416
                S58511_HR_plus_16 S58512_HR_plus_6 S58513_HR_plus_14
ENSG00000000419            14.970           15.930            21.426
ENSG00000000457             9.062           17.188            14.663
ENSG00000000460             7.962           10.645             9.546
ENSG00000000938             8.091            2.126             2.583
ENSG00000000971            27.941           16.921            23.358
ENSG00000001036            10.453           10.133            17.493
                S58514_HR_plus_8 S58515_HR_plus_4 S58516_HR_plus_17
ENSG00000000419           11.599           11.850            16.812
ENSG00000000457            7.321           10.692            14.875
ENSG00000000460            4.911           11.589            14.096
ENSG00000000938            1.736            1.336             1.961
ENSG00000000971           37.848           11.976            22.101
ENSG00000001036            9.405           10.980            10.451
                S58517_HR_plus_20 S58518_HR_plus_9 S58519_HR_plus_10
ENSG00000000419            53.205           17.825            29.516
ENSG00000000457            12.896            6.319            11.145
ENSG00000000460            11.050            8.279            12.118
ENSG00000000938             3.512            1.835             2.116
ENSG00000000971            25.633            9.949            15.823
ENSG00000001036            27.184           13.515             9.778
                S58520_HR_plus_18 S58521_TNBC_1 S58522_HR_plus_2 S58523_TNBC_20
ENSG00000000419            15.273        12.016           12.962         24.139
ENSG00000000457             8.772         9.592           14.977         21.786
ENSG00000000460             4.784         4.546            9.811         15.434
ENSG00000000938             2.567         3.601            1.827          2.035
ENSG00000000971            11.777        20.137           14.854         10.350
ENSG00000001036            16.456         6.935           11.340         18.781
                S58524_HR_plus_19 S58525_TNBC_11 S58526_HR_plus_1 S58527_TNBC_5
ENSG00000000419            18.132         15.714           20.079        29.948
ENSG00000000457            21.983         11.301            8.248         7.567
ENSG00000000460             9.569         25.275            7.739         4.963
ENSG00000000938             1.909          1.451            1.849         4.663
ENSG00000000971            11.330         12.860           17.652         7.739
ENSG00000001036            12.295         11.489           13.692        26.801
                S58528_TNBC_9 S58529_TNBC_8 S58530_TNBC_3 S58531_TNBC_16
ENSG00000000419        30.380        18.034        16.579         25.020
ENSG00000000457         5.265         8.222         9.869          8.186
ENSG00000000460         5.765         2.741        16.369         20.724
ENSG00000000938         1.700         1.804         3.033          4.196
ENSG00000000971        14.067        14.591        12.694         25.955
ENSG00000001036        11.584        20.379        15.991         19.500
                S58532_TNBC_19 S58533_TNBC_4 S58534_TNBC_15 S58535_TNBC_6
ENSG00000000419         21.919        32.646         27.337        25.210
ENSG00000000457          6.449         8.220         11.896         9.801
ENSG00000000460         10.847        16.365         36.008        15.100
ENSG00000000938          1.284         6.876          4.722         3.936
ENSG00000000971          9.392        12.067         16.110        12.524
ENSG00000001036         11.514        12.989         13.420        16.544
                S58536_HR_plus_3 S58537_TNBC_10 S58538_HR_plus_13
ENSG00000000419           11.581         41.396            18.793
ENSG00000000457            7.765          7.715             9.971
ENSG00000000460            4.117         26.625            10.651
ENSG00000000938            2.623          1.857             3.902
ENSG00000000971           12.184          8.468            15.154
ENSG00000001036            6.580         32.567            14.343
                S58539_TNBC_17 S58540_TNBC_2 S58541_TNBC_13 S58542_TNBC_18
ENSG00000000419         15.352        14.655         17.143         37.387
ENSG00000000457         10.145        11.180          9.085         10.207
ENSG00000000460          8.070        18.558          8.689         24.020
ENSG00000000938          3.722         2.894          1.368          3.445
ENSG00000000971          6.359         9.880          4.981         10.279
ENSG00000001036         20.134        10.816         20.089         20.651
                S58543_TNBC_12 S58544_TNBC_7 S60426_HR_plus_27 S60427_TNBC_45
ENSG00000000419         40.120        36.053            33.185         47.930
ENSG00000000457          3.647         6.329            10.811          9.571
ENSG00000000460         12.251         9.676            23.297         18.978
ENSG00000000938          1.672         4.704             1.042          4.295
ENSG00000000971          4.931        11.110             8.357         14.524
ENSG00000001036          8.205        28.994            11.418         20.514
                S60428_HLT_11 S60429_TNBC_48 S60430_TNBC_25 S60431_TNBC_41
ENSG00000000419        31.906         66.317         46.522         28.948
ENSG00000000457        14.893         12.630         10.274          7.784
ENSG00000000460        16.072         27.306         16.943          8.456
ENSG00000000938         1.644          2.561          2.764          5.037
ENSG00000000971        14.728         26.945         14.503         26.263
ENSG00000001036         7.743         28.801         40.385         20.337
                S60432_TNBC_22 S60433_TNBC_29 S60434_TNBC_23 S60435_TNBC_49
ENSG00000000419         30.254         69.403         50.886         32.540
ENSG00000000457         20.943          8.725         17.080          8.518
ENSG00000000460         42.113         13.773         34.933         13.609
ENSG00000000938          3.749          4.359          2.006          5.021
ENSG00000000971         18.958         16.954         14.262         31.358
ENSG00000001036         12.050         19.733         11.672         21.808
                S60436_TNBC_42 S60437_TNBC_28 S60438_TNBC_47 S60439_TNBC_27
ENSG00000000419         33.948         31.757         21.974         36.334
ENSG00000000457          9.172          9.430         14.852          7.988
ENSG00000000460         11.678          8.755         19.622         11.197
ENSG00000000938          3.236          3.736          6.889          3.042
ENSG00000000971         31.367         15.241         24.269         16.827
ENSG00000001036         20.173         34.868         13.536         16.109
                S60440_TNBC_37 S60441_HR_plus_39 S60442_HR_plus_34
ENSG00000000419         44.056            36.186            20.467
ENSG00000000457          8.851             8.698            14.202
ENSG00000000460         10.787             6.871            10.017
ENSG00000000938          1.790             6.430             1.307
ENSG00000000971         60.488            25.704            25.397
ENSG00000001036         20.426            19.041            14.849
                S60443_HR_plus_32 S60444_HR_plus_48 S60445_HLT_2
ENSG00000000419            16.319            14.248       23.661
ENSG00000000457             5.050            10.998       12.519
ENSG00000000460             8.934             5.498        8.997
ENSG00000000938             0.902             1.499        6.910
ENSG00000000971            13.605            31.755       25.167
ENSG00000001036             4.159            15.118        5.705
                S60446_HR_plus_40 S60447_TNBC_24 S60448_HR_plus_47
ENSG00000000419            16.863         13.281            25.526
ENSG00000000457            13.562          8.728            12.199
ENSG00000000460             9.003          4.863             7.080
ENSG00000000938             3.806          6.217             1.120
ENSG00000000971            32.200         62.912            13.368
ENSG00000001036            14.040         15.734            11.848
                S60449_TNBC_38 S60450_TNBC_39 S60451_HR_plus_29 S60452_TNBC_32
ENSG00000000419         24.710         26.753            14.991         33.266
ENSG00000000457         12.872         16.421            11.811          8.975
ENSG00000000460         12.080         14.321            17.220          8.586
ENSG00000000938          0.364          5.660             3.050          4.345
ENSG00000000971          9.273         30.921            12.954         38.284
ENSG00000001036         12.255         18.322            10.808         15.308
                S60453_HR_plus_50 S60454_HR_plus_31 S60455_HR_plus_21
ENSG00000000419            21.575            22.741             2.595
ENSG00000000457            16.864            13.315            12.688
ENSG00000000460            11.155             8.937             4.275
ENSG00000000938             0.800             2.594            11.153
ENSG00000000971            29.931            11.685            10.323
ENSG00000001036            13.006             8.687             6.829
                S60456_HR_plus_28 S60457_HR_plus_26 S60458_HR_plus_35
ENSG00000000419            17.539            16.967            26.874
ENSG00000000457            30.634             7.261            13.376
ENSG00000000460            16.824             6.227             9.806
ENSG00000000938             2.075             1.497             3.499
ENSG00000000971            13.315            11.421           105.264
ENSG00000001036            13.238            15.409            14.832
                S60459_HR_plus_30 S60460_TNBC_50 S60461_TNBC_35 S60462_TNBC_46
ENSG00000000419            16.515         23.686         34.233         29.079
ENSG00000000457            20.835         17.303         10.360         15.735
ENSG00000000460            13.780         11.530         18.096         22.686
ENSG00000000938             3.133          8.277          4.549          5.506
ENSG00000000971            27.406         56.253         30.797         23.591
ENSG00000001036            16.230         14.154         22.347         21.394
                S60463_HR_plus_42 S60464_HLT_28 S60465_HLT_14 S60466_HR_plus_38
ENSG00000000419            23.832        14.799        16.709            21.548
ENSG00000000457            13.704        11.583        12.699            10.685
ENSG00000000460            13.870         8.427        10.986             3.450
ENSG00000000938             4.790         3.064         1.393             2.527
ENSG00000000971            59.416        42.609        24.540            25.219
ENSG00000001036            16.066         6.850         9.554            12.927
                S60467_HR_plus_36 S60468_TNBC_33 S60469_TNBC_30 S60470_HLT_24
ENSG00000000419            46.140         61.823         33.958        24.740
ENSG00000000457            10.000          7.107         10.356         9.858
ENSG00000000460            10.134         17.013         17.238         6.117
ENSG00000000938             2.193          3.470          3.337         1.793
ENSG00000000971            17.962         79.169         26.963        91.862
ENSG00000001036            30.225         40.586         20.326        16.618
                S60471_TNBC_43 S60472_HLT_47 S60473_TNBC_44 S60474_HR_plus_25
ENSG00000000419         38.731        24.260         19.873            17.083
ENSG00000000457         10.052        12.899         12.594            30.481
ENSG00000000460          9.158        10.381          3.775            28.139
ENSG00000000938          4.673         1.810         19.001             0.991
ENSG00000000971         36.290        44.593         18.009             9.900
ENSG00000001036         29.887        14.197         16.274            15.583
                S60475_TNBC_40 S60476_HR_plus_22 S60477_HR_plus_24
ENSG00000000419         29.642            25.160            25.795
ENSG00000000457         10.860             8.712            16.166
ENSG00000000460         12.198            14.640            12.955
ENSG00000000938          3.462             3.868             1.679
ENSG00000000971         19.032            19.086            18.197
ENSG00000001036         17.151            24.155            26.091
                S60478_HLT_38 S60479_TNBC_34 S60480_HLT_22 S60481_HR_plus_37
ENSG00000000419        18.819         19.516        17.739            18.960
ENSG00000000457         9.350          8.280        12.178            12.534
ENSG00000000460         7.591          5.942         6.596             9.717
ENSG00000000938         5.793          2.743         3.422             3.589
ENSG00000000971        72.474         30.864        40.062            23.228
ENSG00000001036        15.921         15.506        11.707            16.254
                S60482_HLT_37 S60483_HLT_44 S60484_HLT_7 S60485_HR_plus_23
ENSG00000000419        22.063        17.104       27.625            17.515
ENSG00000000457         8.242        11.845       15.125            14.210
ENSG00000000460         9.677        12.485        9.750            11.015
ENSG00000000938         2.281         3.042        1.216             2.239
ENSG00000000971        19.155        89.774       34.665            12.453
ENSG00000001036        12.926        13.116       18.029            18.628
                S60486_HLT_17 S60487_HLT_18 S60488_TNBC_31 S60489_HR_plus_49
ENSG00000000419        12.593        21.822         24.156            30.425
ENSG00000000457        10.874         9.494          8.167            12.777
ENSG00000000460         7.474         9.868          5.217             9.045
ENSG00000000938         2.938         5.655          8.403             2.506
ENSG00000000971        33.619        91.411         35.585            44.278
ENSG00000001036        12.213        19.785         18.723            16.014
                S60490_HLT_13 S60491_HLT_1 S60492_HLT_26 S60493_HLT_50
ENSG00000000419        18.181       30.863        17.105        15.130
ENSG00000000457        12.611       12.439        12.114         7.245
ENSG00000000460         9.411        9.824         7.945         9.280
ENSG00000000938         4.665        3.828         3.331         6.950
ENSG00000000971       101.403       63.035        24.098        43.358
ENSG00000001036        15.688       17.347        18.363        13.638
                S60494_HLT_20 S60495_HLT_30 S60496_HLT_31 S60497_HLT_46
ENSG00000000419        21.169        25.750        26.474        24.398
ENSG00000000457        12.445        10.603        10.266         8.156
ENSG00000000460         9.335         8.213        13.066         8.558
ENSG00000000938         2.678         4.059         3.451         3.711
ENSG00000000971        62.067       118.320        68.620       113.366
ENSG00000001036        19.439        18.891        22.397        19.168
                S60498_HLT_21 S60499_HLT_23 S60500_HLT_25 S60502_HLT_43
ENSG00000000419        13.348        11.473        15.971        13.962
ENSG00000000457        10.372        14.886         9.992        12.655
ENSG00000000460         8.248         8.343         8.322         8.575
ENSG00000000938         2.524         2.199         4.732         2.945
ENSG00000000971        23.613        29.124        43.729        36.660
ENSG00000001036        14.096        13.408        18.690        19.183
                S60503_HLT_10 S60504_HLT_16 S60505_HLT_6 S60506_HLT_45
ENSG00000000419        12.939        21.364       33.377        18.779
ENSG00000000457        10.830        18.078        9.695        16.162
ENSG00000000460         8.174        14.451        5.418        10.710
ENSG00000000938         5.222         2.954       11.578         3.098
ENSG00000000971        41.447        30.023       84.563        65.865
ENSG00000001036        16.626        13.424       25.606        22.776
                S60507_HR_plus_33 S60508_HLT_4 S60509_HLT_36 S60510_HLT_34
ENSG00000000419            27.652       15.282        21.124        33.828
ENSG00000000457            13.328       14.554        15.203         8.950
ENSG00000000460             6.458        7.345         9.512        22.860
ENSG00000000938            30.324        5.995         2.616         3.331
ENSG00000000971            61.260       84.052        80.892       126.425
ENSG00000001036            23.476        9.671        20.659        19.097
                S60511_HLT_19 S60512_HLT_48 S60513_HLT_5 S60514_HLT_40
ENSG00000000419        25.992        16.825       19.754        15.477
ENSG00000000457        14.305        14.764       11.329         5.351
ENSG00000000460         9.312        12.759       10.133         3.082
ENSG00000000938         3.494         1.888        6.552         7.523
ENSG00000000971        52.209        59.337       55.525        99.070
ENSG00000001036        19.174        13.194       21.997        19.738
                S60515_HLT_35 S60516_HLT_42 S60517_HLT_33 S60518_HLT_49
ENSG00000000419        27.225        25.124        18.007        27.018
ENSG00000000457         9.150        23.132         7.079         7.067
ENSG00000000460        10.433         7.776         4.226         5.200
ENSG00000000938         5.726         2.874        15.584         4.400
ENSG00000000971        47.765        73.077        34.965        73.633
ENSG00000001036        22.126        26.885        22.343        23.194
                S60519_HLT_3 S60520_HLT_41 S60521_HLT_8 S60522_HLT_12
ENSG00000000419       15.378        28.473       25.967        27.374
ENSG00000000457        7.955        11.921       11.922         9.862
ENSG00000000460        3.228         7.188        8.387        16.840
ENSG00000000938        5.033         2.997        3.718         7.976
ENSG00000000971       71.771        94.688      100.806        42.177
ENSG00000001036       10.340        21.320       18.852        15.498
                S60523_HLT_29 S60524_HLT_39 S60525_HLT_27 S60526_HLT_32
ENSG00000000419        19.238        10.322        10.533        33.387
ENSG00000000457         9.386         6.497        15.148        10.942
ENSG00000000460         9.891         4.058        13.214        15.252
ENSG00000000938         9.557         1.213         6.090         2.156
ENSG00000000971        58.766        58.235        79.803       161.293
ENSG00000001036        16.555        15.616        15.802        20.316
                S60527_HR_plus_45 S60528_HLT_15 S60529_HR_plus_43
ENSG00000000419            14.421        34.877            35.435
ENSG00000000457            12.816        10.537            10.836
ENSG00000000460             6.373         6.757            11.985
ENSG00000000938             4.328        14.365             2.094
ENSG00000000971            40.232       125.532            17.616
ENSG00000001036            21.858        22.167            24.430
                S60530_TNBC_36 S60531_HR_plus_44 S60532_TNBC_26
ENSG00000000419         43.135            30.710         32.038
ENSG00000000457         10.716            10.184          7.474
ENSG00000000460         31.721             7.071         12.697
ENSG00000000938          2.513             2.739          3.459
ENSG00000000971         12.361            31.465         26.623
ENSG00000001036         13.683            15.601         16.784
                S60533_HR_plus_41 S60534_TNBC_21 S60535_HR_plus_46
ENSG00000000419            17.089         26.399            27.712
ENSG00000000457            12.740         11.011            11.618
ENSG00000000460            12.713          6.171             7.259
ENSG00000000938             6.236          0.325             0.609
ENSG00000000971            21.938         56.728            16.429
ENSG00000001036             9.940          9.431            14.625</code></pre>
<pre class="r"><code># gene symbol x patients # Matrix is a numerical matrix with gene symbol as rownames
matrix=data
head(matrix)</code></pre>
<pre><code>                S58505_HR_plus_7 S58506_HR_plus_15 S58507_HR_plus_12
ENSG00000000419           15.916            13.281            15.728
ENSG00000000457           14.932            18.384            19.453
ENSG00000000460           11.570            11.798             9.876
ENSG00000000938            3.312             3.145             4.254
ENSG00000000971           27.985            43.312            43.208
ENSG00000001036           15.597            14.116            15.654
                S58508_HR_plus_5 S58509_TNBC_14 S58510_HR_plus_11
ENSG00000000419           15.069         33.159            31.519
ENSG00000000457           17.914          9.918            12.143
ENSG00000000460           10.387         10.864             9.063
ENSG00000000938            3.592          6.579             1.621
ENSG00000000971           35.365         32.746            14.032
ENSG00000001036           16.926         34.472            13.416
                S58511_HR_plus_16 S58512_HR_plus_6 S58513_HR_plus_14
ENSG00000000419            14.970           15.930            21.426
ENSG00000000457             9.062           17.188            14.663
ENSG00000000460             7.962           10.645             9.546
ENSG00000000938             8.091            2.126             2.583
ENSG00000000971            27.941           16.921            23.358
ENSG00000001036            10.453           10.133            17.493
                S58514_HR_plus_8 S58515_HR_plus_4 S58516_HR_plus_17
ENSG00000000419           11.599           11.850            16.812
ENSG00000000457            7.321           10.692            14.875
ENSG00000000460            4.911           11.589            14.096
ENSG00000000938            1.736            1.336             1.961
ENSG00000000971           37.848           11.976            22.101
ENSG00000001036            9.405           10.980            10.451
                S58517_HR_plus_20 S58518_HR_plus_9 S58519_HR_plus_10
ENSG00000000419            53.205           17.825            29.516
ENSG00000000457            12.896            6.319            11.145
ENSG00000000460            11.050            8.279            12.118
ENSG00000000938             3.512            1.835             2.116
ENSG00000000971            25.633            9.949            15.823
ENSG00000001036            27.184           13.515             9.778
                S58520_HR_plus_18 S58521_TNBC_1 S58522_HR_plus_2 S58523_TNBC_20
ENSG00000000419            15.273        12.016           12.962         24.139
ENSG00000000457             8.772         9.592           14.977         21.786
ENSG00000000460             4.784         4.546            9.811         15.434
ENSG00000000938             2.567         3.601            1.827          2.035
ENSG00000000971            11.777        20.137           14.854         10.350
ENSG00000001036            16.456         6.935           11.340         18.781
                S58524_HR_plus_19 S58525_TNBC_11 S58526_HR_plus_1 S58527_TNBC_5
ENSG00000000419            18.132         15.714           20.079        29.948
ENSG00000000457            21.983         11.301            8.248         7.567
ENSG00000000460             9.569         25.275            7.739         4.963
ENSG00000000938             1.909          1.451            1.849         4.663
ENSG00000000971            11.330         12.860           17.652         7.739
ENSG00000001036            12.295         11.489           13.692        26.801
                S58528_TNBC_9 S58529_TNBC_8 S58530_TNBC_3 S58531_TNBC_16
ENSG00000000419        30.380        18.034        16.579         25.020
ENSG00000000457         5.265         8.222         9.869          8.186
ENSG00000000460         5.765         2.741        16.369         20.724
ENSG00000000938         1.700         1.804         3.033          4.196
ENSG00000000971        14.067        14.591        12.694         25.955
ENSG00000001036        11.584        20.379        15.991         19.500
                S58532_TNBC_19 S58533_TNBC_4 S58534_TNBC_15 S58535_TNBC_6
ENSG00000000419         21.919        32.646         27.337        25.210
ENSG00000000457          6.449         8.220         11.896         9.801
ENSG00000000460         10.847        16.365         36.008        15.100
ENSG00000000938          1.284         6.876          4.722         3.936
ENSG00000000971          9.392        12.067         16.110        12.524
ENSG00000001036         11.514        12.989         13.420        16.544
                S58536_HR_plus_3 S58537_TNBC_10 S58538_HR_plus_13
ENSG00000000419           11.581         41.396            18.793
ENSG00000000457            7.765          7.715             9.971
ENSG00000000460            4.117         26.625            10.651
ENSG00000000938            2.623          1.857             3.902
ENSG00000000971           12.184          8.468            15.154
ENSG00000001036            6.580         32.567            14.343
                S58539_TNBC_17 S58540_TNBC_2 S58541_TNBC_13 S58542_TNBC_18
ENSG00000000419         15.352        14.655         17.143         37.387
ENSG00000000457         10.145        11.180          9.085         10.207
ENSG00000000460          8.070        18.558          8.689         24.020
ENSG00000000938          3.722         2.894          1.368          3.445
ENSG00000000971          6.359         9.880          4.981         10.279
ENSG00000001036         20.134        10.816         20.089         20.651
                S58543_TNBC_12 S58544_TNBC_7 S60426_HR_plus_27 S60427_TNBC_45
ENSG00000000419         40.120        36.053            33.185         47.930
ENSG00000000457          3.647         6.329            10.811          9.571
ENSG00000000460         12.251         9.676            23.297         18.978
ENSG00000000938          1.672         4.704             1.042          4.295
ENSG00000000971          4.931        11.110             8.357         14.524
ENSG00000001036          8.205        28.994            11.418         20.514
                S60428_HLT_11 S60429_TNBC_48 S60430_TNBC_25 S60431_TNBC_41
ENSG00000000419        31.906         66.317         46.522         28.948
ENSG00000000457        14.893         12.630         10.274          7.784
ENSG00000000460        16.072         27.306         16.943          8.456
ENSG00000000938         1.644          2.561          2.764          5.037
ENSG00000000971        14.728         26.945         14.503         26.263
ENSG00000001036         7.743         28.801         40.385         20.337
                S60432_TNBC_22 S60433_TNBC_29 S60434_TNBC_23 S60435_TNBC_49
ENSG00000000419         30.254         69.403         50.886         32.540
ENSG00000000457         20.943          8.725         17.080          8.518
ENSG00000000460         42.113         13.773         34.933         13.609
ENSG00000000938          3.749          4.359          2.006          5.021
ENSG00000000971         18.958         16.954         14.262         31.358
ENSG00000001036         12.050         19.733         11.672         21.808
                S60436_TNBC_42 S60437_TNBC_28 S60438_TNBC_47 S60439_TNBC_27
ENSG00000000419         33.948         31.757         21.974         36.334
ENSG00000000457          9.172          9.430         14.852          7.988
ENSG00000000460         11.678          8.755         19.622         11.197
ENSG00000000938          3.236          3.736          6.889          3.042
ENSG00000000971         31.367         15.241         24.269         16.827
ENSG00000001036         20.173         34.868         13.536         16.109
                S60440_TNBC_37 S60441_HR_plus_39 S60442_HR_plus_34
ENSG00000000419         44.056            36.186            20.467
ENSG00000000457          8.851             8.698            14.202
ENSG00000000460         10.787             6.871            10.017
ENSG00000000938          1.790             6.430             1.307
ENSG00000000971         60.488            25.704            25.397
ENSG00000001036         20.426            19.041            14.849
                S60443_HR_plus_32 S60444_HR_plus_48 S60445_HLT_2
ENSG00000000419            16.319            14.248       23.661
ENSG00000000457             5.050            10.998       12.519
ENSG00000000460             8.934             5.498        8.997
ENSG00000000938             0.902             1.499        6.910
ENSG00000000971            13.605            31.755       25.167
ENSG00000001036             4.159            15.118        5.705
                S60446_HR_plus_40 S60447_TNBC_24 S60448_HR_plus_47
ENSG00000000419            16.863         13.281            25.526
ENSG00000000457            13.562          8.728            12.199
ENSG00000000460             9.003          4.863             7.080
ENSG00000000938             3.806          6.217             1.120
ENSG00000000971            32.200         62.912            13.368
ENSG00000001036            14.040         15.734            11.848
                S60449_TNBC_38 S60450_TNBC_39 S60451_HR_plus_29 S60452_TNBC_32
ENSG00000000419         24.710         26.753            14.991         33.266
ENSG00000000457         12.872         16.421            11.811          8.975
ENSG00000000460         12.080         14.321            17.220          8.586
ENSG00000000938          0.364          5.660             3.050          4.345
ENSG00000000971          9.273         30.921            12.954         38.284
ENSG00000001036         12.255         18.322            10.808         15.308
                S60453_HR_plus_50 S60454_HR_plus_31 S60455_HR_plus_21
ENSG00000000419            21.575            22.741             2.595
ENSG00000000457            16.864            13.315            12.688
ENSG00000000460            11.155             8.937             4.275
ENSG00000000938             0.800             2.594            11.153
ENSG00000000971            29.931            11.685            10.323
ENSG00000001036            13.006             8.687             6.829
                S60456_HR_plus_28 S60457_HR_plus_26 S60458_HR_plus_35
ENSG00000000419            17.539            16.967            26.874
ENSG00000000457            30.634             7.261            13.376
ENSG00000000460            16.824             6.227             9.806
ENSG00000000938             2.075             1.497             3.499
ENSG00000000971            13.315            11.421           105.264
ENSG00000001036            13.238            15.409            14.832
                S60459_HR_plus_30 S60460_TNBC_50 S60461_TNBC_35 S60462_TNBC_46
ENSG00000000419            16.515         23.686         34.233         29.079
ENSG00000000457            20.835         17.303         10.360         15.735
ENSG00000000460            13.780         11.530         18.096         22.686
ENSG00000000938             3.133          8.277          4.549          5.506
ENSG00000000971            27.406         56.253         30.797         23.591
ENSG00000001036            16.230         14.154         22.347         21.394
                S60463_HR_plus_42 S60464_HLT_28 S60465_HLT_14 S60466_HR_plus_38
ENSG00000000419            23.832        14.799        16.709            21.548
ENSG00000000457            13.704        11.583        12.699            10.685
ENSG00000000460            13.870         8.427        10.986             3.450
ENSG00000000938             4.790         3.064         1.393             2.527
ENSG00000000971            59.416        42.609        24.540            25.219
ENSG00000001036            16.066         6.850         9.554            12.927
                S60467_HR_plus_36 S60468_TNBC_33 S60469_TNBC_30 S60470_HLT_24
ENSG00000000419            46.140         61.823         33.958        24.740
ENSG00000000457            10.000          7.107         10.356         9.858
ENSG00000000460            10.134         17.013         17.238         6.117
ENSG00000000938             2.193          3.470          3.337         1.793
ENSG00000000971            17.962         79.169         26.963        91.862
ENSG00000001036            30.225         40.586         20.326        16.618
                S60471_TNBC_43 S60472_HLT_47 S60473_TNBC_44 S60474_HR_plus_25
ENSG00000000419         38.731        24.260         19.873            17.083
ENSG00000000457         10.052        12.899         12.594            30.481
ENSG00000000460          9.158        10.381          3.775            28.139
ENSG00000000938          4.673         1.810         19.001             0.991
ENSG00000000971         36.290        44.593         18.009             9.900
ENSG00000001036         29.887        14.197         16.274            15.583
                S60475_TNBC_40 S60476_HR_plus_22 S60477_HR_plus_24
ENSG00000000419         29.642            25.160            25.795
ENSG00000000457         10.860             8.712            16.166
ENSG00000000460         12.198            14.640            12.955
ENSG00000000938          3.462             3.868             1.679
ENSG00000000971         19.032            19.086            18.197
ENSG00000001036         17.151            24.155            26.091
                S60478_HLT_38 S60479_TNBC_34 S60480_HLT_22 S60481_HR_plus_37
ENSG00000000419        18.819         19.516        17.739            18.960
ENSG00000000457         9.350          8.280        12.178            12.534
ENSG00000000460         7.591          5.942         6.596             9.717
ENSG00000000938         5.793          2.743         3.422             3.589
ENSG00000000971        72.474         30.864        40.062            23.228
ENSG00000001036        15.921         15.506        11.707            16.254
                S60482_HLT_37 S60483_HLT_44 S60484_HLT_7 S60485_HR_plus_23
ENSG00000000419        22.063        17.104       27.625            17.515
ENSG00000000457         8.242        11.845       15.125            14.210
ENSG00000000460         9.677        12.485        9.750            11.015
ENSG00000000938         2.281         3.042        1.216             2.239
ENSG00000000971        19.155        89.774       34.665            12.453
ENSG00000001036        12.926        13.116       18.029            18.628
                S60486_HLT_17 S60487_HLT_18 S60488_TNBC_31 S60489_HR_plus_49
ENSG00000000419        12.593        21.822         24.156            30.425
ENSG00000000457        10.874         9.494          8.167            12.777
ENSG00000000460         7.474         9.868          5.217             9.045
ENSG00000000938         2.938         5.655          8.403             2.506
ENSG00000000971        33.619        91.411         35.585            44.278
ENSG00000001036        12.213        19.785         18.723            16.014
                S60490_HLT_13 S60491_HLT_1 S60492_HLT_26 S60493_HLT_50
ENSG00000000419        18.181       30.863        17.105        15.130
ENSG00000000457        12.611       12.439        12.114         7.245
ENSG00000000460         9.411        9.824         7.945         9.280
ENSG00000000938         4.665        3.828         3.331         6.950
ENSG00000000971       101.403       63.035        24.098        43.358
ENSG00000001036        15.688       17.347        18.363        13.638
                S60494_HLT_20 S60495_HLT_30 S60496_HLT_31 S60497_HLT_46
ENSG00000000419        21.169        25.750        26.474        24.398
ENSG00000000457        12.445        10.603        10.266         8.156
ENSG00000000460         9.335         8.213        13.066         8.558
ENSG00000000938         2.678         4.059         3.451         3.711
ENSG00000000971        62.067       118.320        68.620       113.366
ENSG00000001036        19.439        18.891        22.397        19.168
                S60498_HLT_21 S60499_HLT_23 S60500_HLT_25 S60502_HLT_43
ENSG00000000419        13.348        11.473        15.971        13.962
ENSG00000000457        10.372        14.886         9.992        12.655
ENSG00000000460         8.248         8.343         8.322         8.575
ENSG00000000938         2.524         2.199         4.732         2.945
ENSG00000000971        23.613        29.124        43.729        36.660
ENSG00000001036        14.096        13.408        18.690        19.183
                S60503_HLT_10 S60504_HLT_16 S60505_HLT_6 S60506_HLT_45
ENSG00000000419        12.939        21.364       33.377        18.779
ENSG00000000457        10.830        18.078        9.695        16.162
ENSG00000000460         8.174        14.451        5.418        10.710
ENSG00000000938         5.222         2.954       11.578         3.098
ENSG00000000971        41.447        30.023       84.563        65.865
ENSG00000001036        16.626        13.424       25.606        22.776
                S60507_HR_plus_33 S60508_HLT_4 S60509_HLT_36 S60510_HLT_34
ENSG00000000419            27.652       15.282        21.124        33.828
ENSG00000000457            13.328       14.554        15.203         8.950
ENSG00000000460             6.458        7.345         9.512        22.860
ENSG00000000938            30.324        5.995         2.616         3.331
ENSG00000000971            61.260       84.052        80.892       126.425
ENSG00000001036            23.476        9.671        20.659        19.097
                S60511_HLT_19 S60512_HLT_48 S60513_HLT_5 S60514_HLT_40
ENSG00000000419        25.992        16.825       19.754        15.477
ENSG00000000457        14.305        14.764       11.329         5.351
ENSG00000000460         9.312        12.759       10.133         3.082
ENSG00000000938         3.494         1.888        6.552         7.523
ENSG00000000971        52.209        59.337       55.525        99.070
ENSG00000001036        19.174        13.194       21.997        19.738
                S60515_HLT_35 S60516_HLT_42 S60517_HLT_33 S60518_HLT_49
ENSG00000000419        27.225        25.124        18.007        27.018
ENSG00000000457         9.150        23.132         7.079         7.067
ENSG00000000460        10.433         7.776         4.226         5.200
ENSG00000000938         5.726         2.874        15.584         4.400
ENSG00000000971        47.765        73.077        34.965        73.633
ENSG00000001036        22.126        26.885        22.343        23.194
                S60519_HLT_3 S60520_HLT_41 S60521_HLT_8 S60522_HLT_12
ENSG00000000419       15.378        28.473       25.967        27.374
ENSG00000000457        7.955        11.921       11.922         9.862
ENSG00000000460        3.228         7.188        8.387        16.840
ENSG00000000938        5.033         2.997        3.718         7.976
ENSG00000000971       71.771        94.688      100.806        42.177
ENSG00000001036       10.340        21.320       18.852        15.498
                S60523_HLT_29 S60524_HLT_39 S60525_HLT_27 S60526_HLT_32
ENSG00000000419        19.238        10.322        10.533        33.387
ENSG00000000457         9.386         6.497        15.148        10.942
ENSG00000000460         9.891         4.058        13.214        15.252
ENSG00000000938         9.557         1.213         6.090         2.156
ENSG00000000971        58.766        58.235        79.803       161.293
ENSG00000001036        16.555        15.616        15.802        20.316
                S60527_HR_plus_45 S60528_HLT_15 S60529_HR_plus_43
ENSG00000000419            14.421        34.877            35.435
ENSG00000000457            12.816        10.537            10.836
ENSG00000000460             6.373         6.757            11.985
ENSG00000000938             4.328        14.365             2.094
ENSG00000000971            40.232       125.532            17.616
ENSG00000001036            21.858        22.167            24.430
                S60530_TNBC_36 S60531_HR_plus_44 S60532_TNBC_26
ENSG00000000419         43.135            30.710         32.038
ENSG00000000457         10.716            10.184          7.474
ENSG00000000460         31.721             7.071         12.697
ENSG00000000938          2.513             2.739          3.459
ENSG00000000971         12.361            31.465         26.623
ENSG00000001036         13.683            15.601         16.784
                S60533_HR_plus_41 S60534_TNBC_21 S60535_HR_plus_46
ENSG00000000419            17.089         26.399            27.712
ENSG00000000457            12.740         11.011            11.618
ENSG00000000460            12.713          6.171             7.259
ENSG00000000938             6.236          0.325             0.609
ENSG00000000971            21.938         56.728            16.429
ENSG00000001036             9.940          9.431            14.625</code></pre>
<pre class="r"><code>head(annotation)</code></pre>
<pre><code>                Gene  GeneSymbol Chr start   end   band cytoband isCentromer
2330 ENSG00000223972     DDX11L1   1 12010 13670 p36.33       1p          no
2672 ENSG00000227232      WASH7P   1 14696 24886 p36.33       1p          no
3868 ENSG00000243485 MIR1302-2HG   1 28589 31109 p36.33       1p          no
3700 ENSG00000237613     FAM138A   1 34553 37595 p36.33       1p          no
4172 ENSG00000268020      OR4G4P   1 52473 53312 p36.33       1p          no
3804 ENSG00000240361     OR4G11P   1 62949 63887 p36.33       1p          no
     Position new_positions
2330  12840.0             1
2672  19791.0             2
3868  29849.0             3
3700  36074.0             4
4172  52892.5             5
3804  63418.0             6</code></pre>
<pre class="r"><code>genes=sub(&quot;\\..*&quot;, &quot;&quot;, rownames(matrix))
length(genes)</code></pre>
<pre><code>[1] 48643</code></pre>
<pre class="r"><code>length(unique(genes))</code></pre>
<pre><code>[1] 48643</code></pre>
<pre class="r"><code>gene_map &lt;- setNames(annotation$GeneSymbol, annotation$Gene)
symbols &lt;- gene_map[genes]
sum(is.na(symbols))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>sum(duplicated(symbols[!is.na(symbols)]))</code></pre>
<pre><code>[1] 12070</code></pre>
<pre class="r"><code>summary_df &lt;- data.frame(
  Total = length(symbols),
  Unique = length(unique(symbols)),
  NAs = sum(is.na(symbols)),
  Duplicated = sum(duplicated(symbols[!is.na(symbols)])),
  Empty = sum(symbols == &quot;&quot; | symbols == &quot; &quot;, na.rm = TRUE)
)
print(summary_df)</code></pre>
<pre><code>  Total Unique NAs Duplicated Empty
1 48643  36573   0      12070 12064</code></pre>
<pre class="r"><code>symbols &lt;- make.unique(as.character(symbols))
matrix=apply(matrix, 2, as.numeric)
rownames(matrix)=symbols

matrix=matrix[!is.na(rownames(matrix)), ]
matrix=matrix[rownames(matrix)!=&quot; &quot;, ]
matrix=matrix[rownames(matrix)!=&quot;&quot;, ]


matrix[1:10,1:3]</code></pre>
<pre><code>       S58505_HR_plus_7 S58506_HR_plus_15 S58507_HR_plus_12
DPM1             15.916            13.281            15.728
SCYL3            14.932            18.384            19.453
FIRRM            11.570            11.798             9.876
FGR               3.312             3.145             4.254
CFH              27.985            43.312            43.208
FUCA2            15.597            14.116            15.654
GCLC             27.593            18.909            22.520
NFYA             29.826            33.545            29.806
STPG1             1.920             3.094             2.464
NIPAL3           23.345            17.738            17.147</code></pre>
<pre class="r"><code>consensus=ConsensusTME::consensusTMEAnalysis(matrix, cancerType = &#39;BRCA&#39;)</code></pre>
<pre><code>Producing ConsensusTME Estimates Using The Following Parameters:
 Statistical Framework: &quot;ssgsea&quot;
 Gene Sets For Cancer Type: &quot;BRCA&quot;
 Sample Size: 149</code></pre>
<pre><code>Estimating ssGSEA scores for 19 gene sets.
[1] &quot;Calculating ranks...&quot;
[1] &quot;Calculating absolute values from ranks...&quot;
  |                                                                              |                                                                      |   0%  |                                                                              |                                                                      |   1%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   4%  |                                                                              |===                                                                   |   5%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  17%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |===============                                                       |  21%  |                                                                              |================                                                      |  22%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |=================                                                     |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |===================                                                   |  28%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |=======================                                               |  34%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |==========================                                            |  38%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  40%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |===============================                                       |  44%  |                                                                              |===============================                                       |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |=================================                                     |  48%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  54%  |                                                                              |=======================================                               |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  57%  |                                                                              |========================================                              |  58%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  62%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  64%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |===============================================                       |  68%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |=====================================================                 |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |======================================================                |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  80%  |                                                                              |========================================================              |  81%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |===========================================================           |  84%  |                                                                              |===========================================================           |  85%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |===============================================================       |  91%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================|  99%  |                                                                              |======================================================================| 100%

[1] &quot;Normalizing...&quot;</code></pre>
<pre class="r"><code>consensus=t(consensus)
consensus= merge(consensus, sample_info, by = &quot;row.names&quot;, all = TRUE )</code></pre>
<pre class="r"><code>plot(consensus$Immune_Score, consensus$aneuploidy_score)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(ggplot2)
library(ggpubr)    # For stat_cor()
library(patchwork) # For layout

target_var &lt;- &quot;aneuploidy_score&quot;
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots &lt;- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = &quot;condition&quot;)) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;blue&quot;) +
    stat_cor(method = &quot;pearson&quot;, label.x.npc = &quot;left&quot;, label.y.npc = &quot;top&quot;, size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10)
})

# Combine into a grid
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>target_var &lt;- &quot;ENSG00000085117&quot;
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots &lt;- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = &quot;condition&quot;)) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;blue&quot;) +
    stat_cor(method = &quot;pearson&quot;, label.x.npc = &quot;left&quot;, label.y.npc = &quot;top&quot;, size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10)
})

# Combine into a grid
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df=consensus
df=df[!is.na(df$ENSG00000085117), ]
df$CD82_levels=ifelse(df$ENSG00000085117&gt;median(df$ENSG00000085117), &#39;high-CD82&#39;, &#39;low-CD82&#39;)
df$aneuploidy_score_levels=ifelse(df$aneuploidy_score&gt;median(df$aneuploidy_score), &#39;high-aneuploidy_score&#39;, &#39;low-aneuploidy_score&#39;)

condition_col &lt;- &quot;condition&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>condition_col &lt;- &quot;CD82_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>condition_col &lt;- &quot;aneuploidy_score_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>condition_col &lt;-&quot;CD82_levels&quot; # &quot;CATEGORY.1.&quot; #&quot;CD82_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df[df$condition==&quot;TNBC&quot;,], aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>condition_col &lt;- &quot;aneuploidy_score_levels&quot;
boxplot_vars = colnames(consensus)[2:20]

plots &lt;- lapply(boxplot_vars, function(var) {
  ggplot(df[df$condition==&quot;TNBC&quot;,], aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = &quot;none&quot;)
})
wrap_plots(plots, ncol = 4)</code></pre>
<p><img src="figure/02_deconvolution.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#write.csv(consensus, &#39;/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/consensusTME.csv&#39;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.3.0         ggpubr_0.6.0            ggplot2_3.5.2          
[4] ConsensusTME_0.0.1.9000 RSQLite_2.3.4          

loaded via a namespace (and not attached):
  [1] DBI_1.2.3                   bitops_1.0-9               
  [3] GSEABase_1.62.0             rlang_1.1.2                
  [5] magrittr_2.0.3              git2r_0.32.0               
  [7] matrixStats_1.1.0           compiler_4.3.1             
  [9] mgcv_1.9-0                  DelayedMatrixStats_1.22.6  
 [11] png_0.1-8                   vctrs_0.6.3                
 [13] stringr_1.5.1               pkgconfig_2.0.3            
 [15] crayon_1.5.3                fastmap_1.2.0              
 [17] backports_1.5.0             XVector_0.40.0             
 [19] labeling_0.4.3              promises_1.3.2             
 [21] rmarkdown_2.29              graph_1.78.0               
 [23] purrr_1.0.4                 bit_4.0.5                  
 [25] xfun_0.52                   zlibbioc_1.46.0            
 [27] cachem_1.1.0                beachmat_2.16.0            
 [29] GenomeInfoDb_1.36.4         jsonlite_2.0.0             
 [31] blob_1.2.4                  later_1.4.2                
 [33] rhdf5filters_1.12.1         DelayedArray_0.26.7        
 [35] Rhdf5lib_1.22.1             BiocParallel_1.34.2        
 [37] broom_1.0.8                 irlba_2.3.5.1              
 [39] parallel_4.3.1              R6_2.6.1                   
 [41] bslib_0.9.0                 stringi_1.8.7              
 [43] RColorBrewer_1.1-3          car_3.1-3                  
 [45] GenomicRanges_1.52.1        jquerylib_0.1.4            
 [47] Rcpp_1.0.12                 SummarizedExperiment_1.30.2
 [49] knitr_1.50                  GSVA_1.48.3                
 [51] IRanges_2.34.1              splines_4.3.1              
 [53] httpuv_1.6.16               Matrix_1.6-1               
 [55] tidyselect_1.2.1            rstudioapi_0.17.1          
 [57] dichromat_2.0-0.1           abind_1.4-8                
 [59] yaml_2.3.10                 codetools_0.2-19           
 [61] lattice_0.21-8              tibble_3.2.1               
 [63] Biobase_2.60.0              withr_3.0.2                
 [65] KEGGREST_1.40.1             evaluate_1.0.3             
 [67] Biostrings_2.68.1           pillar_1.10.2              
 [69] carData_3.0-5               MatrixGenerics_1.12.3      
 [71] whisker_0.4.1               stats4_4.3.1               
 [73] generics_0.1.4              rprojroot_2.0.4            
 [75] RCurl_1.98-1.14             S4Vectors_0.38.2           
 [77] sparseMatrixStats_1.12.2    scales_1.4.0               
 [79] xtable_1.8-4                glue_1.8.0                 
 [81] tools_4.3.1                 ScaledMatrix_1.8.1         
 [83] ggsignif_0.6.4              annotate_1.78.0            
 [85] fs_1.6.6                    XML_3.99-0.16              
 [87] rhdf5_2.44.0                grid_4.3.1                 
 [89] tidyr_1.3.1                 AnnotationDbi_1.62.2       
 [91] SingleCellExperiment_1.22.0 nlme_3.1-162               
 [93] GenomeInfoDbData_1.2.10     BiocSingular_1.16.0        
 [95] HDF5Array_1.28.1            Formula_1.2-5              
 [97] cli_3.6.2                   rsvd_1.0.5                 
 [99] workflowr_1.7.1             S4Arrays_1.0.6             
[101] dplyr_1.1.3                 gtable_0.3.6               
[103] rstatix_0.7.2               sass_0.4.10                
[105] digest_0.6.37               BiocGenerics_0.46.0        
[107] farver_2.1.2                memoise_2.0.1              
[109] htmltools_0.5.8.1           lifecycle_1.0.4            
[111] httr_1.4.7                  bit64_4.0.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
