<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tumour Micro-Environment Cell Estimation with ConsensusTME</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Aneuploidy inference</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.research.ieo.it/research-and-technology/technological-units/data-science-unit/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Tumour Micro-Environment Cell Estimation
with ConsensusTME</h1>
<h4 class="date">20 June, 2025</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>casper_ss_ma/analysis/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Volumes/scratch/DIMA/piva/casper_ss_ma/
</td>
<td style="text-align:left;">
..
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatreef0e862c6e53f0632bbba9c3dcb6afd54d693a4f6targetblankf0e862ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/f0e862c6e53f0632bbba9c3dcb6afd54d693a4f6" target="_blank">f0e862c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomdimadatasciencecasperssmatreef0e862c6e53f0632bbba9c3dcb6afd54d693a4f6targetblankf0e862ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/dimadatascience/casper_ss_ma/tree/f0e862c6e53f0632bbba9c3dcb6afd54d693a4f6" target="_blank">f0e862c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .DS_Store
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/02_degs_go_aneuploidy_median.Rmd
    Untracked:  analysis/03_degs_go_CD82expr_median.Rmd
    Untracked:  analysis/VennDiagram.2025-06-09_13-53-40.335615.log
    Untracked:  analysis/VennDiagram.2025-06-09_13-54-51.029086.log
    Untracked:  analysis/VennDiagram.2025-06-09_13-55-15.147126.log
    Untracked:  analysis/VennDiagram.2025-06-09_13-56-18.122749.log
    Untracked:  analysis/VennDiagram.2025-06-09_13-56-30.934079.log
    Untracked:  analysis/VennDiagram.2025-06-09_14-18-19.412377.log
    Untracked:  analysis/VennDiagram.2025-06-18_10-28-53.699452.log
    Untracked:  analysis/VennDiagram.2025-06-18_10-37-36.77178.log
    Untracked:  analysis/VennDiagram.2025-06-18_11-32-36.228427.log
    Untracked:  analysis/VennDiagram.2025-06-18_15-38-55.387683.log
    Untracked:  analysis/VennDiagram.2025-06-18_15-48-17.579371.log
    Untracked:  analysis/VennDiagram.2025-06-18_17-18-17.268774.log
    Untracked:  analysis/VennDiagram.2025-06-19_11-11-17.376961.log
    Untracked:  analysis/VennDiagram.2025-06-19_14-52-46.049026.log
    Untracked:  analysis/VennDiagram.2025-06-19_16-40-05.861139.log
    Untracked:  analysis/VennDiagram.2025-06-19_16-40-07.33202.log
    Untracked:  analysis/VennDiagram.2025-06-19_16-40-08.673023.log
    Untracked:  analysis/VennDiagram.2025-06-19_17-50-05.238063.log
    Untracked:  analysis/VennDiagram.2025-06-19_17-50-07.22979.log
    Untracked:  analysis/VennDiagram.2025-06-19_17-50-09.007028.log
    Untracked:  analysis/VennDiagram.2025-06-19_18-48-01.885712.log
    Untracked:  analysis/VennDiagram.2025-06-19_18-48-03.579702.log
    Untracked:  analysis/VennDiagram.2025-06-19_18-48-04.898695.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-18-23.300456.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-18-24.588109.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-18-26.077856.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-50-54.081682.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-50-55.516535.log
    Untracked:  analysis/VennDiagram.2025-06-20_10-50-56.913582.log
    Untracked:  analysis/VennDiagram.2025-06-20_11-10-43.68944.log
    Untracked:  analysis/VennDiagram.2025-06-20_11-10-45.681514.log
    Untracked:  analysis/VennDiagram.2025-06-20_11-10-47.126222.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-19-10.326514.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-19-11.75991.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-19-13.198666.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-29-09.447741.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-29-11.214146.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-29-12.791818.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-44-02.971891.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-44-04.709094.log
    Untracked:  analysis/VennDiagram.2025-06-20_12-44-06.321173.log
    Untracked:  analysis/hsa04064.HLT-HighAS_vs_HLT-LowAS.png
    Untracked:  analysis/hsa04064.HLT-HighCD82_vs_HLT-LowCD82.png
    Untracked:  analysis/hsa04064.HRplus-HighAS_vs_HRplus-LowAS.png
    Untracked:  analysis/hsa04064.HRplus-HighCD82_vs_HRplus-LowCD82.png
    Untracked:  analysis/hsa04064.HRplus_vs_HLT.png
    Untracked:  analysis/hsa04064.TNBC-HighAS_vs_TNBC-LowAS.png
    Untracked:  analysis/hsa04064.TNBC-HighCD82_vs_TNBC-LowCD82.png
    Untracked:  analysis/hsa04064.TNBC_vs_HLT.png
    Untracked:  analysis/hsa04064.TNBC_vs_HRplus.png
    Untracked:  analysis/hsa04064.png
    Untracked:  analysis/hsa04064.xml
    Untracked:  code/
    Untracked:  data/
    Untracked:  degs_HLT-HighAS_vs_HLT-LowAS.csv
    Untracked:  degs_HLT-HighCD82_vs_HLT-LowCD82.csv
    Untracked:  degs_HRplus-HighAS_vs_HRplus-LowAS.csv
    Untracked:  degs_TNBC-HighCD82_vs_TNBC-LowCD82.csv
    Untracked:  output/

Unstaged changes:
    Modified:   analysis/00_casper_analysis.Rmd
    Deleted:    analysis/02_deconvolution.Rmd
    Modified:   analysis/index.Rmd
    Modified:   casper_ss_ma.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/04_deconvolution.Rmd</code>) and
HTML (<code>docs/04_deconvolution.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/dimadatascience/casper_ss_ma/blob/f0e862c6e53f0632bbba9c3dcb6afd54d693a4f6/analysis/04_deconvolution.Rmd" target="_blank">f0e862c</a>
</td>
<td>
annamariapiva
</td>
<td>
2025-06-20
</td>
<td>
new reports
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)</code></pre>
<div id="consensustme" class="section level1">
<h1>ConsensusTME</h1>
<p>To estimate the abundance of tumor-infiltrating immune and stromal
cell populations from bulk RNA-seq data, we used the ConsensusTME R
package. ConsensusTME implements a consensus-based approach that
integrates multiple published immune deconvolution methods to provide
robust and comprehensive cell-type enrichment scores.</p>
<p>The consensus gene sets are applied within a single-sample gene set
enrichment analysis (ssGSEA) framework to compute normalized enrichment
scores (NES) for each cell type. These scores represent the relative
abundance of immune and stromal cell types across samples, not across
cell types within a sample.</p>
<p>It is important to note that these scores allow for comparisons
between samples, but not between different cell types in the same
sample. The Immune Score summarizes the overall immune infiltration
level for each sample and can be used to stratify tumors based on immune
context.</p>
<div id="samples" class="section level3">
<h3>Samples</h3>
<p>To investigate the relationship between TME content, CD82 gene
expression and aneuploidy levels, we stratified samples based on:</p>
<ul>
<li><p>CD82 expression: Samples were divided into high-CD82 and low-CD82
groups using the median expression value of the gene CD82 across all the
conditions as the cutoff.</p></li>
<li><p>Aneuploidy score: Samples were categorized into high-aneuploidy
and low-aneuploidy groups using a fixed threshold of 0.1 on the
aneuploidy score (defined by looking at aneuploidy score distribution
across all the conditions)</p></li>
</ul>
</div>
<div id="method-selection" class="section level3">
<h3>Method selection</h3>
<p>To estimate cell-type or pathway activity from bulk RNA-seq data, we
applied gene set scoring methods available in ConsensusTME, including
ssgsea, singScore, gsva, plage, and zscore.</p>
<ul>
<li><em>ssGSEA (Single Sample GSEA)</em> calculates enrichment scores
per sample by integrating gene ranks across each gene set. It is robust
to noise and does not require normalized input.</li>
<li><em>singScore</em> is a rank-based, non-parametric method similar to
ssGSEA but simpler and faster.</li>
<li><em>GSVA (Gene Set Variation Analysis)</em> estimates gene set
enrichment as a continuous function of gene expression and is ideal for
normalized data such as TPM or VST.</li>
<li><em>PLAGE (Pathway Level Analysis of Gene Expression)</em> uses
principal component analysis (PCA) to summarize gene set activity,
assuming multivariate normality.</li>
<li><em>Z-score</em> computes the standardized average expression of
genes within each gene set.</li>
</ul>
<p>For the following analysis, we used GSVA method and provided TPM
counts matrix as input.</p>
</div>
<div id="results" class="section level3">
<h3>Results</h3>
<p>The output of the ConsensusTME analysis is a matrix of cell-type
scores, where each row represents a sample and each column corresponds
to an estimated abundance or activity of a specific immune or stromal
cell type. The following heatmap represents the results of the
analysis</p>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div
id="correlation-between-tme-enrichment-and-aneuploidy-score-from-casper"
class="section level1">
<h1>Correlation between TME enrichment and aneuploidy score (from
CaSPeR)</h1>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-4-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-between-tme-enrichment-and-cd82-gene-expression"
class="section level1">
<h1>Correlation between TME enrichment and CD82 gene expression</h1>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-5-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="boxplot-of-tme-enrichment-across-conditions"
class="section level1">
<h1>Boxplot of TME enrichment across conditions</h1>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-6-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div
id="boxplot-of-tme-enrichment-across-cd82-gene-levels-stratified-by-median"
class="section level1">
<h1>Boxplot of TME enrichment across CD82 gene levels (stratified by
median)</h1>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-7-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/04_deconvolution.Rmd/unnamed-chunk-7-2.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div
id="boxplot-of-tme-enrichment-across-aneuploidy-score-levels-stratified-by-median"
class="section level1">
<h1>Boxplot of TME enrichment across aneuploidy score levels (stratified
by median)</h1>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-8-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/04_deconvolution.Rmd/unnamed-chunk-8-2.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div
id="boxplot-of-tme-enrichment-across-cd82-expression-categories-by-ihc-in-tnbc"
class="section level1">
<h1>Boxplot of TME enrichment across CD82 expression categories (by IHC)
in TNBC</h1>
<p>Only TNBC samples are included as only 3 TNBC samples are categorized
as high-expression by IHC</p>
<p><img src="figure/04_deconvolution.Rmd/unnamed-chunk-9-1.png" width="100%" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Rome
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] patchwork_1.3.0         ggpubr_0.6.0            ggplot2_3.5.2          
[4] pheatmap_1.0.13         ConsensusTME_0.0.1.9000 RSQLite_2.4.1          

loaded via a namespace (and not attached):
  [1] DBI_1.2.3                   bitops_1.0-9               
  [3] GSEABase_1.62.0             rlang_1.1.6                
  [5] magrittr_2.0.3              git2r_0.36.2               
  [7] matrixStats_1.5.0           compiler_4.3.1             
  [9] mgcv_1.9-1                  DelayedMatrixStats_1.22.6  
 [11] reshape2_1.4.4              png_0.1-8                  
 [13] vctrs_0.6.5                 stringr_1.5.1              
 [15] pkgconfig_2.0.3             crayon_1.5.3               
 [17] fastmap_1.2.0               backports_1.5.0            
 [19] XVector_0.40.0              labeling_0.4.3             
 [21] promises_1.3.3              rmarkdown_2.29             
 [23] graph_1.78.0                purrr_1.0.4                
 [25] bit_4.6.0                   xfun_0.52                  
 [27] zlibbioc_1.46.0             cachem_1.1.0               
 [29] beachmat_2.16.0             GenomeInfoDb_1.36.4        
 [31] jsonlite_2.0.0              blob_1.2.4                 
 [33] later_1.4.2                 rhdf5filters_1.12.1        
 [35] DelayedArray_0.26.7         Rhdf5lib_1.22.1            
 [37] BiocParallel_1.34.2         broom_1.0.8                
 [39] irlba_2.3.5.1               parallel_4.3.1             
 [41] singscore_1.20.0            R6_2.6.1                   
 [43] bslib_0.9.0                 stringi_1.8.7              
 [45] RColorBrewer_1.1-3          limma_3.56.2               
 [47] car_3.1-3                   GenomicRanges_1.52.1       
 [49] jquerylib_0.1.4             Rcpp_1.0.14                
 [51] SummarizedExperiment_1.30.2 knitr_1.50                 
 [53] GSVA_1.48.3                 IRanges_2.34.1             
 [55] splines_4.3.1               tidyselect_1.2.1           
 [57] httpuv_1.6.16               Matrix_1.6-4               
 [59] rstudioapi_0.17.1           abind_1.4-8                
 [61] yaml_2.3.10                 codetools_0.2-20           
 [63] plyr_1.8.9                  lattice_0.22-7             
 [65] tibble_3.3.0                withr_3.0.2                
 [67] Biobase_2.60.0              KEGGREST_1.40.1            
 [69] evaluate_1.0.4              Biostrings_2.68.1          
 [71] pillar_1.10.2               carData_3.0-5              
 [73] MatrixGenerics_1.12.3       whisker_0.4.1              
 [75] stats4_4.3.1                generics_0.1.4             
 [77] rprojroot_2.0.4             RCurl_1.98-1.17            
 [79] S4Vectors_0.38.2            sparseMatrixStats_1.12.2   
 [81] scales_1.4.0                xtable_1.8-4               
 [83] glue_1.8.0                  tools_4.3.1                
 [85] ScaledMatrix_1.8.1          ggsignif_0.6.4             
 [87] locfit_1.5-9.12             annotate_1.78.0            
 [89] fs_1.6.6                    XML_3.99-0.18              
 [91] rhdf5_2.44.0                grid_4.3.1                 
 [93] tidyr_1.3.1                 edgeR_3.42.4               
 [95] AnnotationDbi_1.62.2        SingleCellExperiment_1.22.0
 [97] nlme_3.1-168                GenomeInfoDbData_1.2.10    
 [99] BiocSingular_1.16.0         HDF5Array_1.28.1           
[101] Formula_1.2-5               cli_3.6.5                  
[103] rsvd_1.0.5                  workflowr_1.7.1            
[105] S4Arrays_1.0.6              dplyr_1.1.4                
[107] gtable_0.3.6                rstatix_0.7.2              
[109] sass_0.4.10                 digest_0.6.37              
[111] BiocGenerics_0.46.0         farver_2.1.2               
[113] memoise_2.0.1               htmltools_0.5.8.1          
[115] lifecycle_1.0.4             httr_1.4.7                 
[117] bit64_4.6.0-1              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
