---
title: "Inference of aneuploidy on RNAseq"
author: 
- Zhan Yinxiu
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  workflowr::wflow_html:
    toc_depth: 4
editor_options:
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo       = FALSE,
                      message    = FALSE,
                      warning    = FALSE,
                      cache      = FALSE,
                      autodep    = TRUE,
                      fig.align  = 'center',
                      fig.width  = 9,
                      fig.height = 9,
                      dpi =        300,
                      dev = c("png", "pdf"))
```


```{r libraries}
library(CaSpER)
```


```{r functions}
mydatatable <- function(x, rownames = FALSE){
  DT::datatable(
  x,
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = rownames
)
}

```


```{r parameters}
seed = 900 # make it reproducible
ncontrol = 30 # number of controls
```

# Objective

In the context of the CD82 protein project, which has been linked to both therapy response and aneuploidy levels, we aim to validate these correlations in a patient cohort. To achieve this, we performed RNA sequencing on samples from:

- 50 triple-negative breast cancer (TNBC) patients (characterized by high aneuploidy),
- 50 hormone receptorâ€“positive (HR+) breast cancer patients (characterized by moderate aneuploidy), and
- 50 healthy controls.

Our objective is to investigate whether a correlation exists between CD82 expression levels and aneuploidy scores in these patients. The goal of this report is to determine the aneuploidy score using the RNA-seq data. [CaSpER](https://github.com/akdess/CaSpER)


# Introduction

The RNA-seq data were analyzed using the Illumina DRAGEN Bio-IT platform (version 4.2.4), with the hg38 genome and Gencode v46 transcriptome reference. We followed the protocol detailed in this [guide](https://rpubs.com/akdes/673120) to compute the aneuploidy scores. At the moment we have 49 healthy donor, 50 HR+ patients and 50 TNBC patients.

# CaSpER pipeline

We used the following annotations

- [cytoband](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/cytoBand.txt.gz)
- [centromeres](http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/centromeres.txt.gz)

```{r read_cytoband}
cytoband <- read.delim("/hpcnfs/scratch/SS/zhan/casper/support_data/cytoBand.txt", header=F)
cytoband <- data.frame(V1=gsub("chr", "", cytoband[,1]), V2=cytoband[,2], V3=cytoband[,3], V4=substring(cytoband$V4, 1, 1), stringsAsFactors=F)
start <- do.call(rbind, lapply(split(cytoband$V2, paste0(cytoband$V1, cytoband$V4)), min))
end <- do.call(rbind, lapply(split(cytoband$V3, paste0(cytoband$V1, cytoband$V4)), max))
cytoband <- data.frame(V1=gsub("p", "", gsub("q", "", rownames(start))), V2=start, V3=end, V4=rownames(start), stringsAsFactors=F)
cytoband <- cytoband [as.vector(unlist(sapply(c(1:22, "X"), function(x) which(cytoband$V1 %in% x)))), ]
cytoband$V4[grep("q", cytoband$V4)] <- "q"
cytoband$V4[grep("p", cytoband$V4)] <- "p"
rownames(cytoband) <- NULL
```

```{r generate_annotation}
annotation <- generateAnnotation(id_type="ensembl_gene_id", genes=rownames(yale_meningioma$data), ishg19=FALSE, centromere)
```

```{r read_baf}
loh <- readBAFExtractOutput(path="/hpcnfs/scratch/SS/zhan/casper/bulk", sequencing.type="bulk")
names(loh) <- gsub(".snp", "", names(loh))
```

```{r read_tmp}
files <- list.files(
  path = "/hpcnfs/scratch/SS/zhan/",
  pattern = "quant\\.genes\\.sf$",
  recursive = TRUE,
  full.names = TRUE
)

data = NULL
for(file in files){
  tmp = read.delim(file, row.names = 1)["TPM"]
  colnames(tmp) = gsub(".quant.genes.sf", "", basename(file))
  if(is.null(data)){
    data = tmp
  }else{
    data = merge(data, tmp, by = 0)
    rownames(data) <- data$Row.names  # Set the row names from the "Row.names" column
    data$Row.names <- NULL                  # Remove the now redundant column
  }
}

rownames(data) = unlist(lapply(strsplit(rownames(data), "\\."), "[[", 1))

#remove gene not present in annotation
data = data[rownames(data) %in% annotation$Gene, ]
```




```{r create_casper_object}
loh.name.mapping = data.frame(loh.name = colnames(data), sample.name = colnames(data))
control.sample.ids = names(loh)[grep("HLT", names(loh))]

set.seed(seed)
control.sample.ids = sample(control.sample.ids, ncontrol)

loh.name.mapping = loh.name.mapping[!(loh.name.mapping$loh.name %in% control.sample.ids), ]
loh = loh[names(loh) %in% loh.name.mapping$loh.name]

object <- CreateCasperObject(raw.data=data, loh.name.mapping=loh.name.mapping, sequencing.type="bulk", 
  cnv.scale=3, loh.scale=3, matrix.type="normalized", expr.cutoff=4.5,
  annotation=annotation, method="iterative", loh=loh, filter="median",  
  control.sample.ids=control.sample.ids, cytoband=cytoband, log.transformed = FALSE, genomeVersion ="hg38")
```


The list of samples used as control is presented below


```{r}
mydatatable(data.frame(control.sample.ids))
```


```{r run_capser, echo=FALSE, results="hide", message=FALSE, warning=FALSE, fig.show="hide"}
final.objects <- runCaSpER(object, removeCentromere=T, cytoband=cytoband, method="iterative")
```

```{r collapse_results}
finalChrMat <- extractLargeScaleEvents(final.objects, thr=0.5)
gamma <- 6
all.segments <- do.call(rbind, lapply(final.objects, function(x) x@segments))
segment.summary <- extractSegmentSummary(final.objects)
loss <- segment.summary$all.summary.loss
gain <- segment.summary$all.summary.gain
loh <- segment.summary$all.summary.loh
loss.final <- loss[loss$count>gamma, ]
gain.final <- gain[gain$count>gamma, ]
loh.final <- loh[loh$count>gamma, ]
```

The tables below show loss/gain/loh events in each samples

```{r show_table_loh_gain_loss}
mydatatable(loh.final)
mydatatable(gain.final)
mydatatable(loss.final)
```

Below we show the fraction of genome affected by either CNV gain/loss/loh

```{r comparative_cnv_across_condition}
# tmp = rbind(gain.final, loh.final, loss.final)
tmp = rbind(gain.final, loss.final)
tmp = aggregate(tmp$width, list(tmp$ID), sum)
tmp$x = tmp$x/ sum(cytoband[cytoband$V4=="q", "V3"])
tmp$type = sub("^[^_]+_(.*)_[^_]+$", "\\1", tmp$Group.1)
aneuploidy = tmp
```

```{r boxplot_comparative_cnv_across_condition}
ggplot(tmp, aes(x=type, y=x)) + geom_boxplot() + xlab("") + ylab("Fraction of genome affected")
```

```{r}
tmp = tmp[tmp$type == "TNBC",]
tmp <- tmp[order(tmp$x), ]
tmp$Group.1 <- factor(tmp$Group.1, levels = tmp$Group.1)
```

```{r barplot_gain_loss_combined}
ggplot(tmp, aes(x = Group.1, y = x)) +
  geom_bar(stat = "identity") +
  coord_flip() +  # This flips the axes to make labels more readable
  theme_minimal() +
  labs(x = NULL, y = "Aneuploidity score", title = "Gain and loss combined") +
  theme(
    axis.text.y = element_text(size = 8),  # Adjust text size as needed
    plot.title = element_text(hjust = 0.5)
  )
```

```{r table_gain_loss_combined}
colnames(tmp) = c("sample", "aneuploidy score", "cancer type")
mydatatable(tmp)
```


```{r comparative_cnv_across_condition_pertype}
all = NULL
# cnv_list = list(gain=gain.final, loh=loh.final, loss=loss.final)
cnv_list = list(gain=gain.final, loss=loss.final)

for(name in names(cnv_list)){
  tmp = cnv_list[[name]]
  tmp = aggregate(tmp$width, list(tmp$ID), sum)
  tmp$x = tmp$x/ sum(cytoband[cytoband$V4=="q", "V3"])
  tmp$type = sub("^[^_]+_(.*)_[^_]+$", "\\1", tmp$Group.1)
  tmp$group = name
  all = rbind(all,tmp)
}
```

```{r boxplot_comparative_cnv_across_condition_pertype}
all$group = as.factor(all$group)
ggplot(all, aes(x=type, y=x, col=group)) + geom_boxplot() + xlab("") + ylab("Fraction of genome affected")
```



```{r}
tmp = all[all$type == "TNBC" & all$group=="loss",]
tmp <- tmp[order(tmp$x), ]
tmp$Group.1 <- factor(tmp$Group.1, levels = tmp$Group.1)
```

```{r barplot_loss}
ggplot(tmp, aes(x = Group.1, y = x)) +
  geom_bar(stat = "identity") +
  coord_flip() +  # This flips the axes to make labels more readable
  theme_minimal() +
  labs(x = NULL, y = "Aneuploidity score", title = "Loss") +
  theme(
    axis.text.y = element_text(size = 8),  # Adjust text size as needed
    plot.title = element_text(hjust = 0.5)
  )
```

```{r table_loss}
colnames(tmp) = c("sample", "aneuploidy score", "cancer type", "cnv_type")
mydatatable(tmp)
```

```{r}
tmp = all[all$type == "TNBC" & all$group=="gain",]
tmp <- tmp[order(tmp$x), ]
tmp$Group.1 <- factor(tmp$Group.1, levels = tmp$Group.1)
```

```{r barplot_gain}
ggplot(tmp, aes(x = Group.1, y = x)) +
  geom_bar(stat = "identity") +
  coord_flip() +  # This flips the axes to make labels more readable
  theme_minimal() +
  labs(x = NULL, y = "Aneuploidity score", title = "Gain") +
  theme(
    axis.text.y = element_text(size = 8),  # Adjust text size as needed
    plot.title = element_text(hjust = 0.5)
  )
```


```{r table_gain}
colnames(tmp) = c("sample", "aneuploidy score", "cancer type", "cnv_type")
mydatatable(tmp)
```

# Correlation aneuplodity and CD82 expression

```{r cd82_correlation_splitted_by_type}
gene_id = annotation$Gene[annotation$GeneSymbol == "CD82"]
cd82_expression = t(data[gene_id, ])
expr_aneuploidy = merge(aneuploidy, cd82_expression, by.y=0, by.x=1)

ggplot(expr_aneuploidy, aes(x = x, y = ENSG00000085117, color = type)) +
  geom_point() +  # Scatter plot
  stat_smooth(method = "lm", se = FALSE) +  # Add linear fit without confidence interval
  stat_cor(aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")), method = "spearman", label.x.npc = "left", label.y.npc = "top") +  # Correlation coefficient
  theme_minimal() +
  labs(title = "",
       x = "Aneuploidy score",
       y = "CD82 expression")
```

```{r cd82_correlation_tumor_vs_normal}
tmp = expr_aneuploidy
tmp$tumor = "yes"
tmp[tmp$type=="HLT","tumor"]= "no"
ggplot(tmp, aes(x = x, y = ENSG00000085117, color=tumor)) +
  geom_point() +  # Scatter plot
  stat_smooth(method = "lm", se = FALSE) +  # Add linear fit without confidence interval
  stat_cor(
    method = "spearman",
    aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
    label.x.npc = "left",
    label.y.npc = "top"
  ) +
  theme_minimal() +
  labs(
    title = "",
    x = "Aneuploidy score",
    y = "CD82 expression"
  )
```

```{r trop2_correlation_tumor_vs_normal}

gene_id = annotation$Gene[annotation$GeneSymbol == "TACSTD2"]
trop2_expression = t(data[gene_id, ])
expr_aneuploidy = merge(expr_aneuploidy, trop2_expression, by.x=1, by.y=0)

tmp = expr_aneuploidy
tmp$tumor = "yes"
tmp[tmp$type=="HLT","tumor"]= "no"
ggplot(tmp, aes(x = x, y = ENSG00000184292, color = tumor)) +
  geom_point() +  # Scatter plot
  stat_smooth(method = "lm", se = FALSE) +  # Add linear fit without confidence interval
  stat_cor(
    method = "spearman",
    aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
    label.x.npc = "left",
    label.y.npc = "top"
  ) +
  theme_minimal() +
  labs(
    title = "",
    x = "Aneuploidy score",
    y = "TROP2 expression"
  )
```

```{r trop2_cd82_correlation}
ggplot(tmp, aes(x = ENSG00000085117, y = ENSG00000184292, color = type)) +
  geom_point() +  # Scatter plot
  stat_smooth(method = "lm", se = FALSE) +  # Add linear fit without confidence interval
  stat_cor(
    method = "spearman",
    aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
    label.x.npc = "left",
    label.y.npc = "top"
  ) +
  theme_minimal() +
  labs(
    title = "",
    x = "CD82 expression",
    y = "TROP2 expression"
  )
```

<!-- # ```{r qc_2} -->
<!-- # cd82_expr = t(data[rownames(data) == annotation$Gene[annotation$GeneSymbol=="CD82"], ]) -->
<!-- # merged  = merge(cd82_expr, control, by=0) -->
<!-- # ggplot(merged, aes(x=type, y=ENSG00000085117)) + geom_boxplot() -->
<!-- # ``` -->
