---
title: "Tumour Micro-Environment Cell Estimation with ConsensusTME"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  workflowr::wflow_html:
    toc_depth: 4
    code_folding: hide
editor_options:
  chunk_output_type: console
---


```{r setup}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```


# ConsensusTME

```{r import}
library(RSQLite)
library(ConsensusTME)
library(pheatmap)

# ConsensusTME::cancerAll # BRCA
```

To estimate the abundance of tumor-infiltrating immune and stromal cell populations from bulk RNA-seq data, we used the ConsensusTME R package. ConsensusTME implements a consensus-based approach that integrates multiple published immune deconvolution methods to provide robust and comprehensive cell-type enrichment scores.

The consensus gene sets are applied within a single-sample gene set enrichment analysis (ssGSEA) framework to compute normalized enrichment scores (NES) for each cell type. These scores represent the relative abundance of immune and stromal cell types across samples, not across cell types within a sample.

It is important to note that these scores allow for comparisons between samples, but not between different cell types in the same sample. The Immune Score summarizes the overall immune infiltration level for each sample and can be used to stratify tumors based on immune context.

```{r}
folderpath='/Volumes/scratch/DIMA/piva/casper_ss_ma/'

# sample_info table
sample_info = read.table(paste0(folderpath, "/data/samples_metadata.csv"), header=T, sep = ',', row.names = 1)
rownames(sample_info) <- sample_info$samplename
sample_info$condition <- as.factor(sample_info$condition)
sample_info$batch <- as.factor(sample_info$batches)


# annotation
load(paste0(folderpath, '/data/data.Rdata'))
#head(data) # TPM counts

# convert ENS gene id to Gene symbol
# gene symbol x patients # Matrix is a numerical matrix with gene symbol as rownames
matrix=data
genes=sub("\\..*", "", rownames(matrix))
gene_map <- setNames(annotation$GeneSymbol, annotation$Gene)
symbols <- gene_map[genes]


summary_df <- data.frame(
  Total = length(symbols),
  Unique = length(unique(symbols)),
  NAs = sum(is.na(symbols)),
  Duplicated = sum(duplicated(symbols[!is.na(symbols)])),
  Empty = sum(symbols == "" | symbols == " ", na.rm = TRUE)
)
#print(summary_df)

symbols <- make.unique(as.character(symbols))
matrix=apply(matrix, 2, as.numeric)
rownames(matrix)=symbols

matrix=matrix[!is.na(rownames(matrix)), ]
matrix=matrix[rownames(matrix)!=" ", ]
matrix=matrix[rownames(matrix)!="", ]

#matrix[1:10,1:10]

```

### Samples
To investigate the relationship between TME content, CD82 gene expression and aneuploidy levels, we stratified samples based on:

- CD82 expression: Samples were divided into high-CD82 and low-CD82 groups using the median expression value of the gene CD82 across all the conditions as the cutoff.

- Aneuploidy score: Samples were categorized into high-aneuploidy and low-aneuploidy groups using a fixed threshold of 0.1 on the aneuploidy score (defined by looking at aneuploidy score distribution across all the conditions)



### Method selection
To estimate cell-type or pathway activity from bulk RNA-seq data, we applied gene set scoring methods available in ConsensusTME, including ssgsea, singScore, gsva, plage, and zscore. 

- *ssGSEA (Single Sample GSEA)* calculates enrichment scores per sample by integrating gene ranks across each gene set. It is robust to noise and does not require normalized input.
- *singScore* is a rank-based, non-parametric method similar to ssGSEA but simpler and faster.
- *GSVA (Gene Set Variation Analysis)* estimates gene set enrichment as a continuous function of gene expression and is ideal for normalized data such as TPM or VST.
- *PLAGE (Pathway Level Analysis of Gene Expression)* uses principal component analysis (PCA) to summarize gene set activity, assuming multivariate normality.
- *Z-score* computes the standardized average expression of genes within each gene set.

For the following analysis, we used GSVA method and provided TPM counts matrix as input.


```{r, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
consensus_list=list()
for (sm in c('ssgsea', 'singScore', 'gsva', 'plage', 'plage', 'zscore')){
  consensus_list[[sm]]=ConsensusTME::consensusTMEAnalysis(matrix, cancerType = 'BRCA', statMethod = sm)
}


```

### Results
The output of the ConsensusTME analysis is a matrix of cell-type scores, where each row represents a sample and each column corresponds to an estimated abundance or activity of a specific immune or stromal cell type.
The following heatmap represents the results of the analysis

```{r, fig.width=15, fig.height=10}

# Create a data frame for annotation (must be a data frame with row names matching column names of matrix)
annotation_df <- as.data.frame(sample_info[, "condition", drop = FALSE])
colnames(annotation_df) <- "Condition"
annotation_colors <- list(
  Condition = c(
    "HLT" = "#66c2a5",
    "HRplus" = "#fc8d62",
    "TNBC" = "#8da0cb"
  )
)

pheatmap(
  consensus_list$gsva,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = TRUE,
  show_colnames = TRUE,
  color = colorRampPalette(c("blue", "white", "red"))(50),
  main = "consensusTME",
  annotation_col = annotation_df,
annotation_colors = annotation_colors,
)


consensus=t(consensus_list$gsva)
consensus=merge(consensus, sample_info, by = "row.names", all = TRUE )

```


# Correlation between TME enrichment and aneuploidy score (from CaSPeR)
```{r, fig.width=15, fig.height=15, out.width="100%"}
library(ggplot2)
library(ggpubr)    # For stat_cor()
library(patchwork) # For layout

target_var <- "aneuploidy_score"
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots <- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = "condition")) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = "lm", se = FALSE) +
    stat_cor(method = "pearson", label.x.npc = "left", label.y.npc = "top", size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10)
})

# Combine into a grid
wrap_plots(plots, ncol = 4)


```

# Correlation between TME enrichment and CD82 gene expression

```{r, fig.width=15, fig.height=15, out.width="100%"}


target_var <- "ENSG00000085117"
feature_vars = colnames(consensus)[2:20]
df=consensus

# Create scatterplots with correlation
plots <- lapply(feature_vars, function(var) {
  ggplot(df, aes_string(x = var, y = target_var, color = "condition")) +
    geom_point(alpha = 0.6) +
    geom_smooth(method = "lm", se = FALSE) +
    stat_cor(method = "pearson", label.x.npc = "left", label.y.npc = "top", size = 3.5) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
  labs(y = "CD82 expr")
})

# Combine into a grid
wrap_plots(plots, ncol = 4)

```


# Boxplot of TME enrichment across conditions

```{r, fig.width=15, fig.height=15, out.width="100%"}
df=consensus
df=df[!is.na(df$ENSG00000085117), ]
df$CD82_levels=ifelse( df$ENSG00000085117> median(df$ENSG00000085117), 'high-CD82', 'low-CD82')
df$aneuploidy_score_levels=ifelse(df$aneuploidy_score> 0.1, 'high-aneuploidy_score', 'low-aneuploidy_score')


condition_col <- "condition"
boxplot_vars = colnames(consensus)[2:20]

plots <- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "none")
})
wrap_plots(plots, ncol = 4)


```

# Boxplot of TME enrichment across CD82 gene levels (stratified by median)

```{r, fig.width=15, fig.height=15, out.width="100%"}
condition_col <- "CD82_levels"
boxplot_vars = colnames(consensus)[2:20]

plots <- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_violin(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "none")
})
wrap_plots(plots, ncol = 4)


plots <- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x =  "condition" , y = var, color = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "right")
})
wrap_plots(plots, ncol = 4)



```

# Boxplot of TME enrichment across aneuploidy score levels (stratified by median)

```{r, fig.width=15, fig.height=15, out.width="100%"}
condition_col <- "aneuploidy_score_levels"
boxplot_vars = colnames(consensus)[2:20]

plots <- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_violin(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "none")
})
wrap_plots(plots, ncol = 4)


plots <- lapply(boxplot_vars, function(var) {
  ggplot(df, aes_string(x =  "condition" , y = var, color = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "right")
})
wrap_plots(plots, ncol = 4)


```

# Boxplot of TME enrichment across CD82 expression categories (by IHC) in TNBC
Only TNBC samples are included as only 3 TNBC samples are categorized as high-expression by IHC

```{r, fig.width=15, fig.height=15, out.width="100%"}
condition_col <-"CATEGORY.1." # "CATEGORY.1." #"CD82_levels"
boxplot_vars = colnames(consensus)[2:20]

plots <- lapply(boxplot_vars, function(var) {
  ggplot(df[df$condition=="TNBC",], aes_string(x = condition_col, y = var, fill = condition_col)) +
    geom_boxplot(outlier.shape = NA, alpha = 0.7) +
    geom_jitter(width = 0.2, alpha = 0.4, size = 0.8) +
    ggtitle(var) +
    theme_minimal(base_size = 10) +
    theme(legend.position = "none")
})
wrap_plots(plots, ncol = 4)


```


```{r}
#write.csv(consensus, '/hpcnfs/scratch/DIMA/piva/casper_ss_ma/data/consensusTME.csv')
```


